h/services/delete_user.py

```diff
@@ -19,38 +15,41 @@ def delete(self, user):
         """
         Deletes a user with all their group memberships and annotations.
 
-        Raises UserDeleteError when deletion fails with the appropriate error
-        message.
+        If a user owns groups with collaborators, the user is unassigned
+        as creator but the group persists.
         """
 
         created_groups = self.request.db.query(Group).filter(Group.creator == user)
-        if self._groups_have_anns_from_other_users(created_groups, user):
-            raise UserDeleteError(
-                "Other users have annotated in groups created by this user"
-            )
+        groups_to_unassign_creator = self._groups_that_have_collaborators(
+            created_groups, user
+        )
+        groups_to_delete = list(set(created_groups) - set(groups_to_unassign_creator))
 
         self._delete_annotations(user)
-        self._delete_groups(created_groups)
+        self._delete_groups(groups_to_delete)
+        self._unassign_groups_creator(groups_to_unassign_creator)
         self.request.db.delete(user)
 
-    def _groups_have_anns_from_other_users(self, groups, user):
+    def _groups_that_have_collaborators(self, groups, user):
         """
-        Return `True` if users other than `user` have annotated in `groups`.
+        Return list of groups that have annotations from other users.
```

Comment:
Would you mind expanding this docstring to include types for params and rtype as it helps to be real explicit that we're passing models around here?